docs/extensions/validation-guidelines-for-extensions.md

```diff
@@ -0,0 +1,44 @@
+# Validation Guidelines For Extensions
+
+This document provides general developer guidelines on validation practices and conventions used in the Gardener extensions codebase.
+
+## Validation of Provider Configuration
+
+With the implementation of [GEP-01](../proposals/01-extensibility.md) the cloud provider, operating system and networking specific knowledge is extracted to external extension controllers.
+Gardener itself is provider-agnostic.
+The API resources contain provider configurations.
+The Gardener API Server does not understand these provider configurations and cannot validate them.
+They should be validated by extension admission components.
+
+Provider configuration fields are of type `*runtime.RawExtension`.
+Extension admission components should decode the provider configuration and perform validation of it.
+
+## Validation of Referenced Resources
```

Comment:
Extensions usually have two components, admission and service controller, the former takes care to admit resources in the Garden cluster and the latter in the seed. This document seems to be focusing on the admission component, I think it should be made more inclusive so that it targets also the service component or an additional section should be added to cover it.

Reply:
No, I mean for example webhooks registered [here](https://github.com/gardener/gardener-extension-provider-aws/blob/ff724a801cad2965d7bff769ff0fb64b415968c7/pkg/cmd/options.go#L69-L80).

Reply:
You are right, the document is focusing on the Garden cluster, sorry for the confusion.